packages/botonic-react/src/webchat-app.jsx

```diff
@@ -185,7 +189,31 @@ export class WebchatApp {
     )
   }
 
-  render(dest, optionsAtRuntime = {}) {
-    render(this.getComponent(optionsAtRuntime), dest)
+  async isWebchatVisible({ appId }) {
+    try {
+      const { status } = await HubtypeService.getWebchatVisibility({
+        appId,
+      })
+      if (status === 200) return true
+      return false
+    } catch (e) {
+      return false
+    }
+  }
+
+  async resolveWebchatVisibility(optionsAtRuntime) {
+    let { appId, visibility } = optionsAtRuntime
+    visibility = visibility || this.visibility
+    if (visibility === undefined) {
+      if (await this.isWebchatVisible({ appId })) return true
+    }
+    if (visibility === true) return true
+    if (typeof visibility === 'function' && visibility()) return true
+    return false
+  }
+
+  async render(dest, optionsAtRuntime = {}) {
+    const isVisible = await this.resolveWebchatVisibility(optionsAtRuntime)
+    if (isVisible) render(this.getComponent(optionsAtRuntime), dest)
```

Comment:
in gneeral I prefer returning an explicit null, not sure if it's a common react practice